File: backend/app/schemas/conversation.py
"""Domain schemas for conversation persistence."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class ConversationCreate(BaseModel):
    """Schema for creating a new conversation."""

    user_id: UUID
    title: str | None = None
    status: str = "active"


class Conversation(BaseModel):
    """Schema representing a persisted conversation."""

    conversation_id: UUID
    user_id: UUID
    title: str | None
    status: str
    created_at: datetime
    updated_at: datetime


class MessageCreate(BaseModel):
    """Schema for creating a new message."""

    conversation_id: UUID
    message_sequence: int
    message_type: str  # 'user' | 'assistant'
    content_text: str


class Message(BaseModel):
    """Schema representing a persisted message."""

    message_id: UUID
    conversation_id: UUID
    message_sequence: int
    message_type: str
    content_text: str
    created_at: datetime


class AIResponseCreate(BaseModel):
    """Schema for creating an AI response record."""

    message_id: UUID
    provider_name: str
    model_name: str
    validation_status: str = "pending"
    input_token_count: int | None = None
    output_token_count: int | None = None
    latency_ms: int | None = None


class AIResponse(BaseModel):
    """Schema representing a persisted AI response."""

    ai_response_id: UUID
    message_id: UUID
    provider_name: str
    model_name: str
    validation_status: str
    input_token_count: int | None
    output_token_count: int | None
    latency_ms: int | None
    created_at: datetime


# ============================================================================
# Phase 5.6 — Conversation history API response schemas
# ============================================================================


class ConversationSummary(BaseModel):
    """One conversation in the authenticated user's conversation list."""

    conversation_id: UUID
    title: str | None
    status: str
    created_at: datetime
    updated_at: datetime


class MessageSummary(BaseModel):
    """One persisted message returned by the conversation history API."""

    message_id: UUID
    conversation_id: UUID
    message_sequence: int
    message_type: str
    content_text: str
    created_at: datetime

File: frontend/src/types/conversation.ts
/**
 * Phase 5.7 — Conversation history contract types.
 *
 * Typed mirror of the Phase 5.6 conversation history API boundary:
 *     GET /api/v1/conversations
 *     GET /api/v1/conversations/{conversation_id}/messages
 *
 * These types mirror the locked backend Pydantic response schemas verbatim
 * (snake_case JSON names, UUIDs represented as strings at the HTTP boundary,
 * datetimes as ISO 8601 strings). The backend is authoritative:
 *     backend/app/schemas/conversation.py (ConversationSummary, MessageSummary)
 *
 * Only fields the backend actually returns are declared. `user_id` is
 * intentionally excluded: the history API never exposes it, and the frontend
 * must not expect it.
 */

/**
 * One conversation in the authenticated user's conversation list.
 *
 * Mirrors backend `ConversationSummary` (backend/app/schemas/conversation.py).
 * Conversations are ordered by `updated_at` DESC by the backend.
 */
export interface ConversationSummary {
  conversation_id: string
  title: string | null
  status: string
  created_at: string
  updated_at: string
}

/**
 * One persisted message returned by the conversation history API.
 *
 * Mirrors backend `MessageSummary` (backend/app/schemas/conversation.py).
 * Messages are ordered by `message_sequence` ASC by the backend.
 */
export interface MessageSummary {
  message_id: string
  conversation_id: string
  message_sequence: number
  message_type: string
  content_text: string
  created_at: string
}
